### PR TITLE
API Use updated FormField::validate() signature

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/11422

### CODE REVIEW

File: src/Forms/WildcardDomainField.php [hunk 1]
```diff
@@ -10,22 +11,21 @@ class WildcardDomainField extends TextField
...
-        }
-
-        $validator->validationError(
-            $this->getName(),
-            _t('DomainNameField.INVALID_DOMAIN', 'Invalid domain name'),
-            'validation'
-        );
-        return $this->extendValidationResult(false, $validator);
+        $result = ValidationResult::create();
+        $this->beforeExtending('updateValidate', function () use ($result) {
```
Comment (ChloeSartorelli) [inline src/Forms/WildcardDomainField.php]:
```suggestion
        $this->beforeExtending('updateValidate', function (ValidationResult $result)) {
```

File: src/Forms/WildcardDomainField.php [hunk 2]
```diff
@@ -10,22 +11,21 @@ class WildcardDomainField extends TextField
...
+            if ($this->checkHostname($this->Value())) {
+                return;
+            }
+            $result->addFieldError(
+                $this->getName(),
+                _t('DomainNameField.INVALID_DOMAIN', 'Invalid domain name'),
+                'validation'
+            );
+        });
+        return $result->combineAnd(parent::validate());
```
Comment (ChloeSartorelli) [inline src/Forms/WildcardDomainField.php]:
```suggestion
        return parent::validate();
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Forms/WildcardDomainField.php (hunk 2 above)
Comment (emteknetnz) [reply src/Forms/WildcardDomainField.php]: Updated

File: src/Forms/WildcardDomainField.php (hunk 1 above)
Comment (emteknetnz) [reply src/Forms/WildcardDomainField.php]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM, will merge when all PRs are ready

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**